apps/examples/camera/camera_main.c

```diff
@@ -227,6 +302,153 @@ static void free_buffer(struct v_buffer *buffers, uint8_t bufnum)
 	}
 }
 
+void clear_camera_options()
+{
+	int i = 0;
+	int size = sizeof(ctrls) / sizeof(struct camera_ctrls) - 1;
+	do {
+		ctrls[i].active = 0;
+		ctrls[i].neg = 0;
+		ctrls[i].req = 0;
+		ctrls[i].value = 0;
+		ctrls[i].min = 0;
+	} while (i++ < size);
+}
+
+void get_camera_options(int fd, int flag)
+{
+	int i = 0;
+	int j = 0;
+	int ret = 0;
+	int size = sizeof(ctrls) / sizeof(struct camera_ctrls) - 1;
+	struct v4l2_queryctrl query;
+	struct v4l2_fmtdesc fmtdesc;
+	struct v4l2_frmsizeenum fsize;
+	char buff[128];
+
+	if (flag) {
+		fprintf(stdout, "Supported Camera Options :\n");
+	}
+	/* Control Options */
+	do {
+		query.id = ctrls[i].v4l2_id;
+		query.ctrl_class = ctrls[i].v4l2_class;
+		ret = ioctl(fd, VIDIOC_QUERYCTRL, &query);
+		ctrls[i].active = 0;
```

Comment:
you can put ctrls[i].active = 0; in else case? instead of executing every time in success case

Reply:
Done